Remove mimemagic

tavyy opened the pull request:
### Jira link

HOTT-<TODO>

### What?

We don't need this dependency and it has licensing issues: https://ukgovernmentdfe.slack.com/archives/CAGBHB4JV/p1616668662022200

Rationale for rails dep changes here: rails/rails#41750 (comment)
Relevant govuk-component changes: DFE-Digital/govuk-components#124

mafonso commented on .github/workflows/deploy_dev.yml:
```diff
@@ -0,0 +1,57 @@
+on: [workflow_dispatch]
+jobs:
+  deploy_dev:
+    if: github.ref == 'refs/heads/master'
```
Do we want to limit it to "master" only on dev?

willfish replied:
Feeexed

mafonso reviewed (approved):
LGTM

tavyy committed:
Remove mimemagic

We don't need this dependency

It has licensing issues:
https://ukgovernmentdfe.slack.com/archives/CAGBHB4JV/p1616668662022200

Rationale for rails dep changes here: rails/rails#41750 (comment)
Relevant govuk-component changes: DFE-Digital/govuk-components#124
